src/openms/include/OpenMS/ANALYSIS/OPENSWATH/SpectrumExtractor.h

```diff
@@ -47,18 +49,44 @@ namespace OpenMS
   HERE
 
   */
-  class OPENMS_DLLAPI SpectrumExtractor
+  class OPENMS_DLLAPI SpectrumExtractor :
+    public DefaultParamHandler
   {
 public:
     SpectrumExtractor();
-    ~SpectrumExtractor();
+    virtual ~SpectrumExtractor();
+
+    void setRTWindow(double rt_window);
+    double getRTWindow() const;
+
+    void setMinScore(double min_score);
+    double getMinScore() const;
+
+    void setMinForwardMatch(double min_forward_match);
+    double getMinForwardMatch() const;
+
+    void setMinReverseMatch(double min_reverse_match);
+    double getMinReverseMatch() const;
+
+    void setMZTolerance(double mz_tolerance);
+    double getMZTolerance() const;
+
+    void setMZToleranceUnits(String mz_tolerance_units);
+    String getMZToleranceUnits() const;
+
+    void getDefaultParameters(Param& params);
```

Comment:
You can add in a function e.g., 
void getParameters (Param& params)
{
getDefaultParameters(params);
params.setValue("rt_window",rt_window_);
...
}
that updates the params object before returning it to the user.  

Alternatively, you could add a member "Param params_;" that stores the params_ object that is updated in each setter method.
In that case, getParameters would look like the following: 

void getParameters(Param& params)
{
params = params_
}

Reply:
opened an issue: https://github.com/biosustain/OpenMS/issues/16